### PR TITLE
Update and add information about Terraform

### PR DESCRIPTION
**Pull request by mkorobeinikov**
- update AWS + terraform
index
- add Azure + terraform
vSpgere + terraform

### CODE REVIEW

File: docs/automation/terraform/terraformvSphere.rst [hunk 1]
```diff
@@ -0,0 +1,402 @@
+:lastproofread: 2024-03-03
+
+.. _terraformvSphere:
+
+Deploying vyos in the vSphere infrastructia
```
Comment (rebortg) [inline docs/automation/terraform/terraformvSphere.rst:5]: typo: infrastructia

File: docs/automation/terraform/terraformAZ.rst [hunk 2]
```diff
@@ -0,0 +1,486 @@
+:lastproofread: 2024-03-03
+
+.. _terraformAZ:
+
+Deploying vyos in the Azure cloud
```
Comment (rebortg) [inline docs/automation/terraform/terraformAZ.rst:5]: VyOS

File: docs/automation/terraform/terraformAZ.rst [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+=================================
+With the help of terraforms, you can quickly deploy Vyos-based infrastructure in the Azure cloud. If necessary, the infrastructure can be removed using terraform.
```
Comment (rebortg) [inline docs/automation/terraform/terraformAZ.rst:7]: missing blank line between headline an next text

**Review by rebortg — CHANGES_REQUESTED** [review]
there are some typos and other small stuff. Please try to wrote brand names correctly.

The Steps are look strange in the renderd [page](https://vyos--1305.org.readthedocs.build/en/1305/automation/terraform/terraformvSphere.html). i would prefer to use the sphinx syntax for enumeration and prevent of using "1.1" or "2.2". just use the single number an 2 space indentation. Maybe have the lowest header level on each step section will also help for a better look.

File: docs/automation/terraform/terraformvyos.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+General commands that we will use for running Terraform scripts
+
+
+.. code-block:: none
+
+  #cd /<your folder>         # go to the Terrafom project
+  #terraform init          # install all addons and provider (aws az and so on)
+  #terraform plan          # show wtah is changing
```
[the remark this answers is not in the document]
Comment (mkorobeinikov) [reply docs/automation/terraform/terraformvyos.rst:33]: done

File: docs/automation/terraform/terraformvyos.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+.. _terraformvyos:
+
+Terraform for VyOS
+==================
+
+VyOS supports development infrastructure via Terraform and provisioning via Ansible.
+Terraform allows you to automate the process of deploying instances on many cloud and virtual platforms. 
+In this article, we will look at using terraforms to deploy vyos on platforms - AWS, AZURE, and vSphere.
+More detailed about what is Terraform you can write using the link_.
```
[the remark this answers is not in the document]
Comment (mkorobeinikov) [reply docs/automation/terraform/terraformvyos.rst:11]: done

**Comment by rebortg** [comment]
@mergifyio backport sagitta

**Pull request merged**